docs/apis/group.yaml

```diff
@@ -0,0 +1,327 @@
+                    name: "Hockey"
+                    description: "Hockey group"
+                    parent: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                    users:
+                      - "2cbb94c7-81f1-44e3-8e9e-75e0c3dd36da"
+
+  /groups/{id}:
+    get:
+      tags:
+        - groups
+      summary: Get a group by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Group details
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Group'
+              example:
+                id: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                name: "Sports"
+                description: "Sports group"
+                parent: null
+                users:
+                  - "7a4b1f8e-5c69-4b60-9232-2b0aaf65ef3c"
+                  - "9f1e47d3-0347-4464-9f02-e0bfae02e896"
+                groups:
+                  - "6b1e7b8d-7e19-41eb-8fa2-c0ee5bb67a94"
+        '404':
+          description: Group not found
+
+    put:
+      tags:
+        - groups
+      summary: Update a group by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/UpdateGroupRequest'
+            example:
+              name: "Activities"
+              description: "Activities group"
+              parent: null
+              users:
+                - "7a4b1f8e-5c69-4b60-9232-2b0aaf65ef3c"
+                - "9f1e47d3-0347-4464-9f02-e0bfae02e896"
+                - "2cbb94c7-81f1-44e3-8e9e-75e0c3dd36da"
+              groups:
+                - "6b1e7b8d-7e19-41eb-8fa2-c0ee5bb67a94"
+      responses:
+        '200':
+          description: Group updated
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Group'
+              example:
+                id: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                name: "Activities"
+                description: "Activities group"
+                parent: null
+                users:
+                  - "7a4b1f8e-5c69-4b60-9232-2b0aaf65ef3c"
+                  - "9f1e47d3-0347-4464-9f02-e0bfae02e896"
+                  - "2cbb94c7-81f1-44e3-8e9e-75e0c3dd36da"
+                groups:
+                  - "6b1e7b8d-7e19-41eb-8fa2-c0ee5bb67a94"
+
+    delete:
+      tags:
+        - groups
+      summary: Delete a group by id
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '204':
+          description: Group deleted
+        '409':
```

Comment:
We don't have a 409 conflict scenario here right?

Reply:
Umm.. Ok.. We can consider "Cannot delete group with child groups" case to a bad request.